core/src/main/java/org/apache/calcite/rel/metadata/RelMdCollation.java

```diff
@@ -240,70 +241,19 @@ private RelMdCollation() {}
    * {@link org.apache.calcite.rel.core.Calc}'s collation. */
   public static List<RelCollation> calc(RelMetadataQuery mq, RelNode input,
       RexProgram program) {
-    return program.getCollations(mq.collations(input));
+    final List<RexNode> projects =
+        program
+            .getProjectList()
+            .stream()
+            .map((p) -> program.expandLocalRef(p))
+            .collect(Collectors.toList());
+    return enumrableCalc(mq, input, projects);
```

Comment:
Wouldn't it be easier to leave the project method unchanged, and here just return:
`return project(mq, input, projects);`
?

Reply:
+1, cause there is no other invoke for `enumrableCalc` currently.

Reply:
OK